src/cplscheme/SerialCouplingScheme.cpp

```diff
@@ -93,10 +93,13 @@ void SerialCouplingScheme::performReceiveOfFirstAdvance()
   if (doesFirstStep()) {
     // do nothing
   } else { // second participant
+    // similar to SerialCouplingScheme::exchangeSecondData()
     receiveAndSetTimeWindowSize();
     PRECICE_DEBUG("Receiving data...");
     receiveData(getM2N(), getReceiveData());
-    checkDataHasBeenReceived();
+    if (not hasDataBeenReceived()) { // check is required, because calling checkDataHasBeenReceived() if hasDataBeenReceived() == true would trigger an assertion. This situation can only occur during initialization of the second participant of a SerialCouplingScheme, because it may call receive twice: Once for receiving initial data, once for receiving the result of the first advance.
+      checkDataHasBeenReceived();
```

Comment:
This construction looks odd. Looks a lot like `hasDataBeenReceived` should be set to `false` somewhere.

Reply:
We could do this before `performReceiveOfFirstAdvance()` in `BaseCouplingScheme::initialize`, but then we again need to differentiate between the different coupling schemes, because this only has to be done for the `SerialCouplingScheme`. If we want to set `_hasDataBeenReceived = false` inside of `SerialCouplingScheme::performReceiveOfFirstAdvance()`, we do not really gain a lot and we will have to make `_hasDataBeenReceived` protected (currently it's private).

Be aware that this work-around will also go away as soon as we can remove `performReceiveOfFirstAdvance()` (which will be the case in #1523 when we are able to send and receive the complete storage at once).

Reply:
Actually. Let's just remove `performReceiveOfFirstAdvance` here. It's so much simpler. d73a326c734986c24553ed4fe7731d7da710eacd